### PR TITLE
Create Report Issue Button

### LINKED ISSUE #683: Create a report button
We will add in the task page a new *Report* button in order to notify our team that is a wrong issue or doesn't exist, or even if there's an error.

When a report is sent from the form (as in the screenshot), our notification email (notifications@gitpay.me) should receive an e-mail with the following template:

Subject: We received a new report from an issue imported to Gitpay
```
We received a new report from the task <a href="{{url}}">{{title}}</a>:

Reason: {{selectedReason}}

<a href="{{approveURL}}">Approve</a>
```

Once approved, it should direct to a route that will delete the issue. What would be your solution to this?

One suggestion: In order to delete an issue from a user, we will have to add a token in the task, and when we generate the link by e-mail, it directs to the route we need to create and the token should match in order to delete that task.

The task owner should receive a confirmation that the issue was deleted:
Subject: Your task created on Gitpay was deleted by our team
```
The task you've created on Giptay: <a href="{{url}}">{{title}}</a>

It was deleted by our team cause it violates our guidelines.

Reason: {{selectedReason}}
```

### Why
We're having in production many issues that should be tested, or it was invalid before we implement a better validation, so this will be a tool to help us to moderate using the community for invalid or nonrelevant issues added.

### Level
Beginner

### Current behavior
There's no button on the task page to report an issue

### Expected behavior
A new button will be a

### Useful links
  - https://material-ui.com/

### Screenshot attached
Report button
Report dialog

### How to reproduce the user scenario
 - Create a user
 - Import an issue from Github
 - On the task page from this issue on Gitpay, you should add a Report button

### What you will need to complete this issue
You will need the SENDGRID test keys that will be sent once you're assigned to this issue.

### Tips
- We have existing examples from other actions on the bar that will open a model and send e-mails too that you can have a reference for how to implement this feature.
- Don't forget to use the internationalization API to display the text in different languages

### PR DESCRIPTION
**Pull request by longj724**
## Description

> Created a _Report_ button on the task page that notifies the email _report@gitpay.me_ that there is an issue with the task.

## Changes

- Frontend: added a report button that opens a form for users to send a report based on a selected reason. Created a redux action reportTask that makes a call to the server once the form is sent.
- Backend: Created a route that sends an email to _report@gitpay.me_ and a route to delete the the task that is sent as a url in the report email. Once a task is reported a token is created and attached as a query parameter to the delete route to secure it. 

## Issue

> Closes #683 

## Impacted Area

> https://gitpay.me/#/task/taskId

## Steps to test

Steps needed to reproduce the scenario from this change to validate if the pull request solve this issue

- Create a task with an account
- Log into a different account and report the created task
- _report@gitpay.me_ should receive email to delete the task
- Check the email of the account that created that task to confirm that an email notifying them of the task deletion was sent.

Screen Shot 2020-07-16 at 10 09 41 PM

Screen Shot 2020-07-16 at 10 11 12 PM

Screen Shot 2020-07-16 at 10 12 08 PM

### CODE REVIEW

File: frontend/src/actions/taskActions.js [hunk 1]
```diff
@@ -259,8 +263,25 @@ const syncTaskError = error => {
...
+  return { type: REPORT_TASK_SUCCESS, completed: true }
+}
+
+const reportTaskError = error => {
+  return { type: REPORT_TASK_ERROR, completed: true, error: error }
+}
+
 const createTask = (task, history) => {
-  validToken()
+  // eslint-disable-next-line no-console
```
Comment (alexanmtz) [inline frontend/src/actions/taskActions.js:283]: @longj724 why you removed the token function and added debug, I think you were testing and forgot these lines here.

File: frontend/src/components/task/task-report.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+  static defaultProps = { visible: true }
+
+  static propTypes = {
+    visible: PropTypes.bool,
+    onClose: PropTypes.func,
+    reportTask: PropTypes.func
+  }
+
+  onChangeReason = (e) => {
+    const invalid = 'The issue is invalid, it was removed by the providers (Github, Bitbucket) and it\'s not available anymore'
```
Comment (alexanmtz) [inline frontend/src/components/task/task-report.js:41]: The reasons should be translated string as well, so you can create references like `report.reasons.reason1`

File: frontend/src/components/task/task-report.js [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    const fourOFour = 'This page is an error, for some reason the content is not loaded, or I recieve a 404'
+    const testOrAccident = 'This issue seems to be a test issue, or it was added accidentally'
+
+    if (e.target.value.localeCompare(invalid) === 0 ||
```
Comment (alexanmtz) [inline frontend/src/components/task/task-report.js:45]: Maybe better use array references like `reasons = ['reason1', 'reason2', 'reason3']`

File: frontend/src/translations/result/br.json [hunk 4]
```diff
@@ -542,5 +542,15 @@
   "first.task.community.description": "We will make a campaign to let our community know that you have an issue to be solved",
   "first.task.community.title": "Send to our community",
   "first.task.next": "Next",
-  "first.task.back": "Back"
+  "first.task.back": "Back",
```
Comment (alexanmtz) [inline frontend/src/translations/result/br.json:545]: @longj724 You don't need to change the result files, they're managed by our translation tool, so please don't add the result files on this PR.

File: frontend/src/translations/result/en.json [hunk 5]
```diff
@@ -542,5 +542,15 @@
   "first.task.community.description": "We will make a campaign to let our community know that you have an issue to be solved",
   "first.task.community.title": "Send to our community",
   "first.task.next": "Next",
-  "first.task.back": "Back"
+  "first.task.back": "Back",
```
Comment (alexanmtz) [inline frontend/src/translations/result/en.json:545]: @longj724 Same here. You don't need to change the result files, they're managed by our translation tool, so please don't add the result files on this PR.

File: locales/result/br.json [hunk 6]
```diff
@@ -144,6 +144,10 @@
   "mail.task.instructions": "<p>Como trabalhar neste projeto?</p><ul><li>Acesse a tarefa no Gitpay;</li><li>Clique em \"Estou interessado\";</li><li>Acesse o projeto e crie uma solução;</li><li>Envie um Pull Request com a solução;</li><li>O dono do projeto revisará e você fará os ajustes necessários para que a tarefa atenda os requisitos;</li><li>Receba a recompensa quando a tarefa for concluída.</li></ul>",
   "mail.task.docs.title": "Nós temos esta documentação completa para ajudá-lo (apenas em Português por enquanto)",
   "mail.task.reason": "Você recebeu esta mensagem pois se cadastrou no <a href=\"http://gitpay.me\" target=\"_blank\">Gitpay</a>",
+  "mail.report.send.action": "Recebemos um novo relatório de um problema importado para o Gitpay",
```
Comment (alexanmtz) [inline locales/result/br.json:147]: Same here

File: locales/result/en.json [hunk 7]
```diff
@@ -144,6 +144,10 @@
   "mail.task.instructions": "<p>How to start to work on this project?</p><ul><li>Access the task on Gitpay</li><li>Click \"I'm interested\"</li><li>Access the project repo and create a fork</li><li>Send a Pull Request</li><li>The maintaners will review, then you will receive a proper feedback and you will make the necessary adjustments to fit with the project guidelines</li><li>Then you will be rewarded<br></li></ul>",
   "mail.task.docs.title": "We have this documentation to help you (in Portuguese for now)",
   "mail.task.reason": "You received this message because you subscribed to <a href=\"http://gitpay.me\" target=\"_blank\">Gitpay</a>",
+  "mail.report.send.action": "We received a new report from an issue imported to Gitpay",
```
Comment (alexanmtz) [inline locales/result/en.json:147]: Same here.

File: frontend/src/actions/taskActions.js (hunk 1 above)
Comment (longj724) [reply frontend/src/actions/taskActions.js:283]: Yes that must have been a mistake - will fix this and everything else.

**Comment by longj724** [comment]
When I switch the language to Portuguese from English, none of the report translations work, however when I then refresh the page with the language set as Portuguese, they work fine. Do you know of any reason why this could be happening?

**Comment by alexanmtz** [comment]
@longj724 probably the process locally is not changing correctly the redux state, but once you refresh it loads correctly cause it's stored on LocalStorage.

So don't worry about it, now we have a command `npm run translate` on the frontend folder to generate the source file from your changes, so just use an `id` and a `defaultText` attribute and the script will generate and handle the translation using the tool to generate different languages.

You don't need to worry about it.

**Comment by longj724** [comment]
I think I made all the requested changes - the package-lock.json is still showing up as a changed file, however I ran it through a diff checker and it is no different than the current gitpay master branch. Not sure how to remove it from the pull request.

**Comment by alexanmtz** [comment]
We don't have any conflict on merge, so I think we're good to go, great job @longj724 💯

**Pull request merged**